### PR TITLE
Add odbc2parquet

### PR DESCRIPTION
**Pull request by jonashaag**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jonashaag** [comment]
@pacman82 please let me know if it's ok for you to be a maintainer here.

File: recipes/odbc2parquet/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  number: {{ build_num }}
+  script:
+    - cargo install --root "$PREFIX" --path .
+    - cargo-bundle-licenses --format yaml --output THIRDPARTY.yml
+
+requirements:
+  build:
+    - {{ compiler('c') }}       # [linux]
+    - {{ compiler('rust') }}
+    - unixodbc                  # [unix]
```
Comment (xhochy) [inline recipes/odbc2parquet/meta.yaml:23]: This should not be necessary. Do you get an error if you remove it?

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by pacman82** [comment]
I am fine with being a maintainer here. Happy to help out with issues if I can. I am still a conda noob though

**Comment by jonashaag** [comment]
```
Installing $PREFIX\bin\odbc2parquet.exe
   Installed package `odbc2parquet v2.0.2 (C:\bld\odbc2parquet_1694858318694\work)` (executable `odbc2parquet.exe`)
warning: be sure to add `$PREFIX\bin` to your PATH to be able to run the installed binaries
```

But then it doesn't find `odbc2parquet.exe` in the tests, why?

**Comment by xhochy** [comment]
On Windows you need to install into `%LIBRARY_PREFIX%` which is `%PEEFIX%\Library`.

File: recipes/odbc2parquet/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  version: {{ version }}
+
+source:
+  url: https://github.com/pacman82/odbc2parquet/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 8d635c8407d7e79816f9c2cc102f0cff28876e2ea95f47f82b8e108c534b0846
+
+build:
+  number: {{ build_num }}
+  script:
+    - cargo install --root "$PREFIX" --path .
```
Comment (xhochy) [inline recipes/odbc2parquet/meta.yaml:16]: Other recipes seem to work without this arg (probably set by rust activation)
```suggestion
    - cargo install --path .
```

File: recipes/odbc2parquet/meta.yaml (hunk 1 above)
Comment (jonashaag) [reply recipes/odbc2parquet/meta.yaml:23]: Yes

File: recipes/odbc2parquet/meta.yaml (hunk 2 above)
Comment (jonashaag) [reply recipes/odbc2parquet/meta.yaml]: Doesn't seem to work

**Comment by jonashaag** [comment]
The file is now correctly installed on Windows but it's still not found during the test

File: recipes/odbc2parquet/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+source:
+  url: https://github.com/pacman82/odbc2parquet/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 8d635c8407d7e79816f9c2cc102f0cff28876e2ea95f47f82b8e108c534b0846
+
+build:
+  number: {{ build_num }}
+  script:
+    - cargo install --path . --root "$PREFIX"          # [not win]
+    - cargo install --path . --root "$LIBRARY_PREFIX"  # [win]
```
Comment (xhochy) [inline recipes/odbc2parquet/meta.yaml:17]:
```suggestion
    - cargo install --path . --root "%LIBRARY_PREFIX%"  # [win]
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by xhochy** [comment]
The build logs show that the package is empty on Windows.

File: recipes/odbc2parquet/meta.yaml (hunk 1 above)
Comment (xhochy) [reply recipes/odbc2parquet/meta.yaml:24]: Can you post that? I'm unable to navigate to it on mobile.

Comment (jonashaag) [reply recipes/odbc2parquet/meta.yaml:24]:
```
error: linking with `/home/conda/staged-recipes/build_artifacts/odbc2parquet_1694870537904/_build_env/bin/x86_64-conda-linux-gnu-cc` failed: exit status: 1
  |
  = note: LC_ALL="C" PATH="/home/conda/staged-recipes/build_artifacts/odbc2parquet_1694870537904/_build_env/lib/rustlib/x86_64-unknown-linux-gnu/bin:/home/conda/staged-recipes/build_artifacts/odbc2parquet_1694870537904/_build_env/.cargo/bin:/home/conda/staged-recipes/build_artifacts/odbc2parquet_1694870537904/_build_env/bin:/home/conda/staged-recipes/build_artifacts/odbc2parquet_1694870537904/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/bin:/opt/conda/condabin:/home/conda/staged-recipes/build_artifacts/odbc2parquet_1694870537904/_build_env/bin:/home/conda/staged-recipes/build_artifacts/odbc2parquet_1694870537904/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/bin:/opt/conda/bin:/opt/conda/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/home/conda/bin" VSLANG="1033" "/home/conda/staged-recipes/build_artifacts/odbc2parquet_1694870537904/_build_env/bin/x86_64-conda-linux-gnu-cc" "-m64" "/tmp/rustc3qkB0L/symbols.o" "/home/conda/staged-recipes/build_artifacts/odbc2parquet_1694870537904/work/target/x86_64-unknown-linux-gnu/release/deps/odbc2parquet-0b6faf9750dd8881.odbc2parquet.8e1cb2c0a8aea5b0-cgu.01.rcgu.o" "-Wl,--as-needed" "-L" "/home/conda/staged-recipes/build_artifacts/odbc2parquet_1694870537904/work/target/x86_64-unknown-linux-gnu/release/deps" "-L" "/home/conda/staged-recipes/build_artifacts/odbc2parquet_1694870537904/work/target/release/deps" "-L" "/home/conda/staged-recipes/build_artifacts/odbc2parquet_1694870537904/work/target/x86_64-unknown-linux-gnu/release/build/lz4-sys-a27774d65445ae18/out" "-L" "/home/conda/staged-recipes/build_artifacts/odbc2parquet_1694870537904/work/target/x86_64-unknown-linux-gnu/release/build/zstd-sys-a260d1a5697fa5b2/out" "-L" "/home/conda/staged-recipes/build_artifacts/odbc2parquet_1694870537904/_build_env/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-Wl,-Bstatic" "/tmp/rustc3qkB0L/libzstd_sys-1cb384104906291f.rlib" "/tmp/rustc3qkB0L/liblz4_sys-3ca9b5e41b9d9f44.rlib" "/home/conda/staged-recipes/build_artifacts/odbc2parquet_1694870537904/_build_env/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-b114db70ea0690b1.rlib" "-Wl,-Bdynamic" "-lodbc" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-Wl,--eh-frame-hdr" "-Wl,-z,noexecstack" "-L" "/home/conda/staged-recipes/build_artifacts/odbc2parquet_1694870537904/_build_env/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-o" "/home/conda/staged-recipes/build_artifacts/odbc2parquet_1694870537904/work/target/x86_64-unknown-linux-gnu/release/deps/odbc2parquet-0b6faf9750dd8881" "-Wl,--gc-sections" "-pie" "-Wl,-z,relro,-z,now" "-Wl,-O1" "-nodefaultlibs" "-Wl,-rpath-link,/home/conda/staged-recipes/build_artifacts/odbc2parquet_1694870537904/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib" "-Wl,-rpath,/home/conda/staged-recipes/build_artifacts/odbc2parquet_1694870537904/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib"
  = note: /home/conda/staged-recipes/build_artifacts/odbc2parquet_1694870537904/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/12.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: cannot find -lodbc: No such file or directory
          collect2: error: ld returned 1 exit status
```

Comment (xhochy) [reply recipes/odbc2parquet/meta.yaml:24]: There is some link flag missing. Can you try the following:

```
export RUSTFLAGS="-Clink-args=-Wl,-L$PREFIX/lib"
```

Comment (xhochy) [reply recipes/odbc2parquet/meta.yaml:24]: This will otherwise break later on if you cross-compile.

**Comment by jonashaag** [comment]
@xhochy looks good now

**Review by xhochy — APPROVED** [review]

**Pull request merged**